azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -58,21 +294,36 @@ public OSImage getImage(String id) {
    }
 
    @Override
-   public Iterable<String> listLocations() {
-      // TODO Auto-generated method stub
-      return null;
+   public Iterable<Location> listLocations() {
+      return api.getLocationApi().list();
    }
 
    @Override
-   public Deployment getNode(String id) {
-      // TODO Auto-generated method stub
-      return null;
+   public Deployment getNode(final String id) {
+      return FluentIterable.from(api.getCloudServiceApi().list())
+              .transform(new Function<CloudService, Deployment>() {
+                 @Override
+                 public Deployment apply(CloudService input) {
+                    return api.getDeploymentApiForService(input.name()).get(id);
+                 }
+              })
+              .firstMatch(notNull())
+              .orNull();
    }
 
    @Override
-   public void destroyNode(String id) {
-      // TODO Auto-generated method stub
+   public void destroyNode(final String id) {
+      CloudService cloudService = api.getCloudServiceApi().get(id);
+
+      // TODO detach disk before deleting node
 
+      String deleteDeploymentRequestId = api.getDeploymentApiForService(cloudService.name()).delete(id);
+      String operationDescription = format("Delete deployment(%s) of cloudService(%s)", id, cloudService.name());
+      waitForOperationCompletion(deleteDeploymentRequestId, operationDescription);
+
+      String cloudServiceDeleteRequestId = api.getCloudServiceApi().delete(cloudService.name());
+      operationDescription = format("Delete cloudService(%s)", cloudService.name(), Long.toString(operationTimeout));
+      waitForOperationCompletion(cloudServiceDeleteRequestId, operationDescription);
```

Comment:
When destroying nodes, should we consider removing the storage account at some point?